File: problems/34/solution.py
class Solution:
    result = []

    def postorder(self, root):
        if not root:
            return

        self.postorder(root.left)
        self.postorder(root.right)
        self.result.append(root.val)

    def postorderTraversal(self, root):
        # Implement the solution here
        self.result = []
        self.postorder(root)
        return self.result

File: problems/34/solution.java
// TreeNode type
// class TreeNode {
//     int val;
//     TreeNode left;
//     TreeNode right;
//     TreeNode(int x) { val = x; left = null; right = null; }
// }
class Solution {
    ArrayList<Integer> result = new ArrayList<Integer>();

    public ArrayList<Integer> postorderTraversal(TreeNode root) {
        // Implement the solution
        _postorder(root);
        return result;
    }


    public void _postorder(TreeNode root) {
        if (root == null)
            return;
        _postorder(root.left);
        _postorder(root.right);
        result.add(root.val);
    }

}
